src/Revit/RevitHyparTools/RevitExtensions.cs

```diff
@@ -1,11 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
 using Autodesk.Revit.DB;
 using Elements.Geometry;
 
-namespace Hypar.Revit 
+namespace Hypar.Revit
 {
-    public static class RevitExtensions {
-        public static Vector3 ToVector3(this XYZ xyz) {
+    public static class RevitExtensions
+    {
+        public static Vector3 ToVector3(this XYZ xyz)
+        {
             return new Vector3(xyz.X, xyz.Y, xyz.Z);
-        }       
+        }
+        public static Polygon ToPolygon(this CurveLoop curveLoop)
+        {
+            return new Polygon(curveLoop.Select(l => l.GetEndPoint(0).ToVector3()).ToList());
+        }
+
+        /// <summary>
+        /// Analyzes all of the PlanarFaces of a Revit solid and returns those that 
+        /// face "up" within a certain threshold.
+        /// </summary>
+        /// <param name="solid">The revit Solid</param>
+        /// <param name="verticalThreshold">The angle (in degrees) that represents the threshold for a face to be considered facing up.  A completely horizontal face will have an angle of 0.  A face that does not face up or down at all will have an angle of 90.</param>
+        public static PlanarFace[] GetMostLikelyTopFaces(this Solid solid, double verticalThreshold = 30)
```

Comment:
Do a review of these methods and give them the most limited access possible. For example, this method is called inside one other private method. Does it need to be public?

Reply:
There are lots of these that can be set to internal, is that a worthwhile limitation as well?

Reply:
?

Reply:
Yes. Every public field or method is a field that you have to support. They become part of the public API. Let’s start by hiding them, then expose them when we get feedback that they’re needed.

Reply:
gotcha, makes sense